crates/bevy_sprite/src/mesh2d/mesh.rs

```diff
@@ -72,6 +114,34 @@ impl Plugin for Mesh2dRenderPlugin {
     }
 }
 
+#[cfg(feature = "bevy_shader_hot_reloading")]
+fn set_shader_import_paths(
```

Comment:
I had a go at making this more dry, not sure if I like it, but maybe you're able to make something of it?

```rust
#[cfg(feature = "bevy_shader_hot_reloading")]
struct HotShader {
    strong_handle: Handle<Shader>,
    loaded: bool,
    import_path: String,
}

// Needed to keep the shaders alive
#[cfg(feature = "bevy_shader_hot_reloading")]
struct HotShaders<T> {
    shaders: Vec<HotShader>,
    marker: PhantomData<T>,
}

#[cfg(feature = "bevy_shader_hot_reloading")]
impl<T> Default for HotShaders<T> {
    fn default() -> Self { // can't currently be derived think it relates to this issue: https://github.com/rust-lang/rust/issues/26925
        Self {
            shaders: Default::default(),
            marker: Default::default(),
        }
    }
}

#[cfg(feature = "bevy_shader_hot_reloading")]
fn setup_shader_imports<T: 'static + Send + Sync>(
    mut hot_shaders: ResMut<HotShaders<T>>,
    mut shaders: ResMut<Assets<Shader>>,
    asset_server: Res<AssetServer>,
) {
    for hot_shader in hot_shaders.shaders.iter_mut() {
        if !hot_shader.loaded
            && asset_server.get_load_state(hot_shader.strong_handle.clone()) == LoadState::Loaded
        {
            shaders
                .get_mut(hot_shader.strong_handle.clone())
                .unwrap()
                .set_import_path(&hot_shader.import_path);

            hot_shader.loaded = true;
        }
    }
}
```

```rust
//impl Plugin for Mesh2dRenderPlugin {
            app.insert_resource(HotShaders::<Self>{
                ...
            });
            app.add_system(setup_shader_imports::<Self>);
```

Reply:
I added something like this into bevy_render for the purposes of keeping shaders alive, and for setting their import paths when they have been loaded if they have an import path.